hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractInvocationFuture.java

```diff
@@ -96,9 +97,10 @@
      */
     private volatile Object state = VOID;
 
-    protected AbstractInvocationFuture(Executor defaultExecutor, ILogger logger) {
+    protected AbstractInvocationFuture(Executor defaultExecutor, ILogger logger, boolean disableUnblockableThreadCheck) {
```

Comment:
I think it's better to create client-executor with the default `ThreadFactory` instead of `UnblockablePoolExecutorThreadFactory` when user-code-deployment is enabled. This way, executor threads won't be `UnblockableThread ` threads and assertion will be ignored. 
This `disableUnblockableThreadCheck ` param just for disabling the assertion looks ugly.

Reply:
This makes perfect sense